contracts/Tokens/Prime/Prime.sol

```diff
@@ -615,11 +634,53 @@ contract Prime is IIncomeDestination, AccessControlledV8, PrimeStorageV1 {
         }
     }
 
-    function _getUnderlying(address vToken) internal view returns (address) {
-        if (vToken == vBNB) {
-            return WBNB;
-        } else {
-            return IVToken(vToken).underlying();
-        }
+    //////////////////////////////////////////////////
+    //////////////// APR Calculation ////////////////
+    ////////////////////////////////////////////////    
+
+    function _incomePerBlock(address vToken) internal view returns (uint256) {
+        IVToken market = IVToken(vToken);
+        return ((((market.totalBorrows() * market.borrowRatePerBlock()) / EXP_SCALE) * market.reserveFactorMantissa()) / EXP_SCALE);
+    }
+
+    function _distributionPercentage() internal view returns (uint256) {
+        return IProtocolShareReserve(protocolShareReserve).getPercentageDistribution(
+            address(this), IProtocolShareReserve.Schema.SPREAD_PRIME_CORE
+        );
+    }
+
+    function _incomeDistributionYearly(address vToken) internal view returns (uint256) {
+        uint256 totalIncomePerBlock = _incomePerBlock(vToken);
+        uint256 incomePerBlockForDistribution = (totalIncomePerBlock * _distributionPercentage()) / IProtocolShareReserve(protocolShareReserve).MAX_PERCENT();
+        return BLOCKS_PER_YEAR * incomePerBlockForDistribution;
+    }
+
+    function _calculateUserAPR(
+        address vToken, 
+        address user, 
+        uint256 totalSupply, 
+        uint256 totalBorrow
+    ) internal view returns (uint256 supplyAPR, uint256 borrowAPR) {
+        uint256 userScore = interests[vToken][user].score;
+        uint256 totalScore = markets[vToken].sumOfMembersScore;
+
+        uint256 userYearlyIncome = (userScore * _incomeDistributionYearly(vToken)) / totalScore;
+        uint256 totalValue = totalSupply + totalBorrow;
+
+        uint256 userSupplyIncomeYearly = (userYearlyIncome * totalSupply) / totalValue;
+        uint256 userBorrowIncomeYearly = (userYearlyIncome * totalBorrow) / totalValue;
+
+        supplyAPR = (userSupplyIncomeYearly * MAXIMUM_BPS) / totalSupply;
+        borrowAPR = (userBorrowIncomeYearly * MAXIMUM_BPS) / totalBorrow;
```

Comment:
I think we can simplify this, with something like this:
```
apr = (userYearlyIncome * MAXIMUM_BPS) / totalValue;
supplyAPR = totalSupply > 0 ? apr : 0;
borrowAPR = totalBorrow > 0 ? apr : 0;
```

Reply:
As we talked offline, the allocation of the rewards for the APR should be done considering the capped amounts of the supplied and borrowed amounts. So, this "optimization" doesn't make sense anymore